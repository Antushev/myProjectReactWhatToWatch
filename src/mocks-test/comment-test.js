const comments = [
  {
    id: 1,
    user: `Джон`,
    rating: 7.4,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  },
  {
    id: 2,
    user: `Андрей`,
    rating: 9,
    comment: `Хороший фильм на четыре раза`,
    date: `July 28, 2020`
  },
  {
    id: 3,
    user: `Джон`,
    rating: 8.8,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  },
  {
    id: 4,
    user: `Джон`,
    rating: 1.5,
    comment: `Хороший фильм на один раз`,
    date: `July 28, 2020`
  }
];

export {comments};
