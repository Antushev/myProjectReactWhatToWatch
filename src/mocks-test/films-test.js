import {generateFilm, generateFilms} from '../mocks/films.js';

const FILMS_NUMBER = 10;

const film = generateFilm(1);
const films = generateFilms(FILMS_NUMBER);

export {film, films};
