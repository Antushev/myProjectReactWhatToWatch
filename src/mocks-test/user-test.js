const user = {
  id: 1,
  name: `Джек`,
  email: `jack@sina.com`,
  avatar: `/wtw/static/avatar/8.jpg`
};

export {user};
