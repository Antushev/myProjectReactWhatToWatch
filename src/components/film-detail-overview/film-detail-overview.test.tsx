import * as React from 'react';
import * as renderer from 'react-test-renderer'

import {film} from '../../mocks-test/films-t';

import FilmDetailOverview from './film-detail-overview';

describe(`FilmDetailComponent`, () => {
  it(`FilmDetailComponentSnapshot`, () => {
    const renderTabs = jest.fn();

    const tree = renderer.create(
        <FilmDetailOverview
          film={film}
          renderTabs={renderTabs}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
