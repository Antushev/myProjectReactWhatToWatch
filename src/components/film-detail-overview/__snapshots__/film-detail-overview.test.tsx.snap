// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailComponent FilmDetailComponentSnapshot 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="aviator.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <div
        className="movie-rating"
      >
        <div
          className="movie-rating__score"
        >
          4
        </div>
        <p
          className="movie-rating__meta"
        >
          <span
            className="movie-rating__level"
          >
            Normal
          </span>
          <span
            className="movie-rating__count"
          >
            89543
          </span>
        </p>
      </div>
      <div
        className="movie-card__text"
      >
        <p>
          In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
                    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
        </p>
        <p
          className="movie-card__director"
        >
          <strong>
            Director: 
            Джон
          </strong>
        </p>
        <p
          className="movie-card__starring"
        >
          <strong>
            Starring: 
            Гоша, Щас Зе Енд, Куллер, Райан Оркестр
            and other
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
